include/lbann/layers/layer.hpp

```diff
@@ -29,6 +29,8 @@
 
 #include "lbann/base.hpp"
 #include "lbann/comm.hpp"
+#include "lbann/execution_contexts/execution_context.hpp"
+#include "lbann/execution_contexts/sgd_execution_context.hpp"
```

Comment:
These shouldn't be `#include`d here. They aren't used in this file.

Reply:
Including these headers promotes sloppiness. It's painful to include them in all of the child classes because our child classes don't match our programming model. The pain is good and appropriate.